### PR TITLE
build: migrate to SDK-V2

### PR DESCRIPTION
**Pull request by melsener**
## Change description

Migrate to TeamsJS-V2.

You can review this PR commit-by-commit. 
- The [first commit](https://github.com/zeplin/microsoft-teams-app/pull/200/commits/8dcba7550d991274f098bc208ae2ec94832b62a1) consists of the automatic changes made by Teams Toolkit (see Reference-1) to update the SDK. Along with updating some code, it adds some TODOs for us.
- The later five commits handle those TODOs
- [The next one](https://github.com/zeplin/microsoft-teams-app/pull/200/commits/4f96122aa7f83bb89834345e4f230c5cbe037397) updates the SDK and manifest to the recommended version (see Reference-2)
- [The next commit](https://github.com/zeplin/microsoft-teams-app/pull/200/commits/d41866258bfd4af202402e7355f4044e4168475a) updates the login flow due to the change in the `authentication` method (See Reference-3 and 4)
- [The next commit](https://github.com/zeplin/microsoft-teams-app/pull/200/commits/a8df13cac1b66515740262603a5f833c01ea2149) removes early returns in case of `undefined` since this was also possible in the previous version (see Reference-5 and 6)

References:
1. [Updating via Teams Toolkit](https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/using-teams-client-library?tabs=javascript%2Cmanifest-teams-toolkit#updating-to-teamsjs-version-20:~:text=The%20easiest%20way%20to%20update%20your%20Teams%20app%20with%20TeamsJS%20version%202.0.x%20is%20to%20use%20the%20Teams%20Toolkit%20extension)
2. [Existing teams apps - SDK and Manifest recommended versions](https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/using-teams-client-library?tabs=javascript%2Cmanifest-teams-toolkit#:~:text=Update%20to%20TeamsJS%20v.2.19.0%20when%20possible)
3. [AuthenticateParameters interface on `v1.11.0`](https://github.com/OfficeDev/microsoft-teams-library-js/blob/release-1.11.0/src/public/authentication.ts#L316) and [deprecation warning](https://learn.microsoft.com/en-us/javascript/api/@microsoft/teams-js/authentication.authenticateparameters?view=msteams-client-js-latest)
4. [Authenticate Popup Parameters interface docs](https://learn.microsoft.com/en-us/javascript/api/@microsoft/teams-js/authentication.authenticatepopupparameters?view=msteams-client-js-latest)
5. [Updates to `Context` interface](https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/using-teams-client-library?tabs=javascript%2Cmanifest-teams-toolkit#updates-to-the-context-interface)
6. [Context interface on `v1.11.0`](https://github.com/OfficeDev/microsoft-teams-library-js/blob/release-1.11.0/src/public/interfaces.ts#L170)

## Type of change
- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

### Development

- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [x] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### Code review

- [x] Pull request has a descriptive title and it follows [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) format and breaking change indicator if required (You can use the [Angular convention](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type))
- [ ] Screenshots or screencasts are attached as necessary
- [x] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: src/client/containers/ConfigurationCreateContainer/hooks/useConfigurationCreate.ts [hunk 1]
```diff
@@ -61,13 +61,14 @@ export const useConfigurationCreate = ({
...
     useEffect(() => {
         if (isInitialized) {
-            microsoftTeams.getContext(({
-                channelId,
-                channelName,
-                tid: tenantId
-            }) => {
-                microsoftTeams.settings.getSettings(settings => {
-                    microsoftTeams.settings.registerOnSaveHandler(async saveEvent => {
+            app.getContext().then(({
```
Comment (dorukgunes) [inline src/client/containers/ConfigurationCreateContainer/hooks/useConfigurationCreate.ts:64]: Nothing important but, should we also update to async, await style to improve readability ?

**Review by dorukgunes — COMMENTED** [review]
Did we have a change to test it with old ms teams ?

**Comment by melsener** [comment]
> Did we have a change to test it with old ms teams ?

yes @dorukgunes however, I couldn't test it with new ms-teams and teams cloud due to the issues on their side.

**Review by dorukgunes — APPROVED** [review]

**Review by eren-zeplin — APPROVED** [review]

**Review by aykutbulca — APPROVED** [review]
Let's not merge it until we make sure we can publish a new manifest (aka we have the dx account permissions back)

**Pull request merged**